iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -92,16 +179,19 @@ public ShuntCompensatorAdder setTargetDeadband(double targetDeadband) {
     public ShuntCompensatorImpl add() {
         String id = checkAndGetUniqueId();
         TerminalExt terminal = checkAndGetTerminal();
-        ValidationUtil.checkbPerSection(this, bPerSection);
-        ValidationUtil.checkSections(this, currentSectionCount, maximumSectionCount);
+        if (model == null) {
+            throw new ValidationException(this, "the shunt compensator model has not been defined");
+        }
+        model.getModel(AbstractShuntCompensatorModel.class).checkCurrentSection(this, currentSectionCount);
```

Comment:
model.getModel() is really stange design: is the variable name is the good one? The adder set this variable with a model implementation, so I don't understand why the model is able to return a model...

I think you look to the ReactiveLimitsHolder? The design is a bit different because we instanciate ReactiveLimitsHolderImpl?

Reply:
Yes, I may have not named the objects well, I will change a bit the design in my next commit